player/main.c

```diff
@@ -446,6 +450,11 @@ int mp_initialize(struct MPContext *mpctx, char **options)
     MP_STATS(mpctx, "start init");
 
 #if HAVE_COCOA
+    if (opts->vo->video_driver_list == NULL ||
+        strcmp(opts->vo->video_driver_list->name, "opengl-cb") != 0)
+    {
+        cocoa_cb_dealloc();
+    }
```

Comment:
I think it's possible for the video_driver_list to have 0 items, but still not be a NULL pointer. Then `name` would be NULL.

Also I'm not so happy that this code exists. I assume it's for switching between the legacy backend?

Reply:
i have to pre-allocate the CAOpeGLLayer because the context is created async by it. i have to make sure cocoa-cb is properly closed if not used, otherwise it would leave an invisible background window, an active context and it would react to events.

maybe there is a better place to call this function, somewhere were we can be sure that opengl-cb is not used?

Reply:
We could add an API function that returns whether opengl-cb is connected to an actual VO. Basically this would return whether `p->ctx->active` is set (see init/uninit functions in vo_opengl_cb.c). The update callback gets called when the VO is destroyed, so you could check this when you receive an update notification.

Reply:
maybe i am misunderstanding something, but the update callback is never called when opengl-cb isn't used? so it would always leave cocoa-cb allocated in that case.

Reply:
Yeah, it has the other way around. We need a "do something if this is not going to be used", which is pretty hard, because we have no idea if it's going to be used.